pkg/services/ngalert/schedule/schedule.go

```diff
@@ -139,18 +154,164 @@ func (sch *schedule) Unpause() error {
 }
 
 func (sch *schedule) Run(ctx context.Context) error {
-	sch.wg.Add(1)
+	sch.wg.Add(2)
 
 	go func() {
 		if err := sch.ruleEvaluationLoop(ctx); err != nil {
 			sch.log.Error("failure while running the rule evaluation loop", "err", err)
 		}
 	}()
 
+	go func() {
+		if err := sch.adminConfigSync(ctx); err != nil {
+			sch.log.Error("failure while running the admin configuration sync", "err", err)
+		}
+	}()
+
 	sch.wg.Wait()
 	return nil
 }
 
+// SyncAndApplyConfigFromDatabase looks for the admin configuration in the database and adjusts the sender(s) accordingly.
+func (sch *schedule) SyncAndApplyConfigFromDatabase() error {
+	sch.log.Info("start of admin configuration sync")
+	orgIds, err := sch.adminConfigStore.GetOrgsWithAdminConfiguration()
+	if err != nil {
+		return err
+	}
+
+	q := &models.GetOrgAdminConfiguration{OrgIDs: orgIds}
+	if err := sch.adminConfigStore.GetAdminConfigurations(q); err != nil {
+		return err
+	}
+
+	sch.log.Info("found admin configurations", "count", len(q.Result))
```

Comment:
Should we return early on `len(q.Result) == 0`? Else this long line saying "found" would not make sense

Reply:
We can't return early here.  If you had 10 orgs sending configuration to an alertmanager and suddenly you deleted all of their configs, we'd need to stop them so it needs to go through the whole cycle. The log line would be useful to determine that because we found no configurations we stopped all the senders.

Reply:
Update: I moved the log line to debug at the end of the day.